### DISCUSSION TITLE
Feature request: per project colors

### DISCUSSION THREAD

**Discussion by mstade** [supabase/supabase#1769] [Feature Requests]
Hey team!

So, this may just be by accident but I've enabled the dark theme in one browser window and the light theme in another window and the settings seem to stick even through page refreshes and navigations. If I open another window it'll be whatever theme I selected last. Here's a screenshot of the two windows with different themes enabled:

<img width="1440" alt="Screenshot 2021-05-30 at 23 57 27" src="https://user-images.githubusercontent.com/605731/120121360-49fbc980-c1a3-11eb-962f-96d61cf8b243.png">

We've got two projects set up: one for development (i.e. staging) and another for production (no touchy touchy!) We don't want to mess up the production database (obviously) and the fact that these two windows had different themes enabled actually saved us some work today, because we were about to run a script in the wrong project when the color of the theme made us stop and do a double take on the project name, only to realize we were in the wrong project.

So here's the idea: could we perhaps assign different colors to different projects, that would better disambiguate what project you're working in? I'm thinking the header is a prime target for such colorization:

<img width="1440" alt="Screenshot 2021-05-31 at 00 06 55" src="https://user-images.githubusercontent.com/605731/120121558-664c3600-c1a4-11eb-8daa-86f89a2bbb23.png">
<img width="1440" alt="Screenshot 2021-05-31 at 00 06 46" src="https://user-images.githubusercontent.com/605731/120121578-7d8b2380-c1a4-11eb-8ad3-5546a7fdd892.png">

What'dya think?

**Comment by MildTomato** [comment]
@mstade that's a great idea.

I also remember using Firebase and accidentally changing the settings of the wrong project 🙈  so I know what you're experiencing here.

We were considering adding in different color themes into the dashboard, but I hadn't even considered it on a project level, so I'll keep that in mind when working on it. We're just looking at adding customization to our UI library, and once we have a basic version of that in place, we can start to offer a few different themes for each project as you suggested.

Another possible thing that could help is we highlight the project name in a color of your choosing, or perhaps add a coloured label/icon next to the name so it's clear at a glance which project you are in.

On another note, I've also been trying out different layouts of the top nav bar so that it's more obvious what org and project you are in - I'll take another look at this.

**Reply by mstade** [reply]
Thank you so much for your response, and for considering the idea! 🙏🏻😊

> Another possible thing that could help is we highlight the project name in a color of your choosing, or perhaps add a coloured label/icon next to the name so it's clear at a glance which project you are in.

The bigger the better I'd say. If it's just the project name it might be too small to notice anyway, which is exactly what happened in another tool we use where you can change color of different connections, but it's just a little rectangle in the top which is easily missed. (Sorry I'm on my phone right now but will try to dig out screenshots later.)

Taking this idea a little further, could the color of a project perhaps be something that's automatically set when you create a new one? I.e. make a set of say 10-15 curated colors that are sufficiently distinct (I'm thinking viz tools like [color brewer](https://colorbrewer2.org/) can help here) and just pick the next in line when a new project is created. Of course keeping the option to change the color, but also providing sane defaults.

Anyway, I'm rambling. I'm sure whatever (if anything!) you guys come up with will be stellar, as usual! Thanks for all the great work! 🙏🏻

**Reply by MildTomato** [reply]
Just to keep this updated, and because as all of you have seen; we have not really tackled this yet.

However, we are deeply investigating a better model for environment management that has a much larger scope than this.

We'll see if we can share anything with the community, as feedback would be very useful.

**Comment by probablykasper** [comment]
I think this is a really important feature which most people don't realize they need. It would be huge to have a clear color difference between production and staging/local

**Comment by probablykasper** [comment]
I keep making changes without realizing I'm in production due of this. Please @kiwicopple @joshenlim @MildTomato @saltcod ❤️

**Reply by MildTomato** [reply]
Hey @probablykasper - sorry for radio silence. 

We're weighing up wether to do something like this, or just hold off for a more defined environment support update. It might be that implementing what @mstade originally suggested would just end up getting deprecated quite soon as there might not be any need for it anymore. 

Do you have different projects for each environment. One of prod, one for staging ?

**Reply by probablykasper** [reply]
Thanks for responding! Yes I do have different projects for prod and staging, however it's not just about differentiating prod vs staging, but also the local dev environment.

Getting a more defined environment update sounds like a big task that could take many months to complete, so I would definitely love if this was added in the meantime, even if it does get deprecated later.

An idea that might be easier: Instead of having a bunch of project colors, there could be just 2 themes to choose from, one for prod and dev/staging.

**Reply by mstade** [reply]
How about this: give user's the ability to specify the color of a project. Doesn't have to be much, just the header bar or something. If you've got a fancy design system, make it set the accent color or something. 🤷🏻‍♂️

This is only incidentally useful as a project differentiator – so perhaps doesn't interfere with your grand master plan and is useful even _after_ those fancy features arrive – while still providing huge value for idiots.

In the meantime @probablykasper here's what I've done. For our preview env I use an emoji in the name. It's not ideal, it's too subtle, but it's _something_ at least. For the production env I've not dared using emojis in the name because I dunno – burned so many times by assumptions programmers make about strings I guess...

<img width="1512" alt="image" src="https://user-images.githubusercontent.com/605731/233115833-1c21e7b3-373c-463f-a6d8-d6127772ac3a.png">

**Comment by danielbayley80** [comment]
+1 for this.

I am running multiple projects, and I really worry about my prod accounts ..... that I accidentally do something  via the dashboard.  I am considering running two accounts but that is a bit of a chore, i think at a basic level adding color coding so I can highlight my prod accounts in red or something would help prevent the most stupid of user errors.

Beyond this I think the ability to have resource / project groups and even assign further protections would be very beneficial .

It could be as simple as this example where you pick a primary color for the header. I can then make all my prod projects red, prototypes in blue, or whatever. 

![image](https://github.com/user-attachments/assets/991baeb9-86a0-42d6-a042-ef5a0537844a)

**Comment by maor-outrival** [comment]
+1 for this. 

4 years later and you guys still havent "tackled" this? It's such a simple thing to implement, c'mon guys.
